src/Settings.ts

```diff
@@ -78,6 +78,12 @@ export namespace Settings {
         }
     }
 
+    export namespace UserCustomSetting {
+        export function path(): string | undefined {
+            return workspace.getConfiguration("maven.custom").get<string>("settingFileLocation");

```

Comment:
by specifying scope to "machine-overridable" above, here you should allow to get workspace level setting instead of global one.

Reply:
It will be better to get this config from workspace configuration, if not find then get from global configuration?